ecs-cli/integ/cmd_up/up_test.go

```diff
@@ -0,0 +1,289 @@
+			continue
+		}
+
+		// Then
+		if ok := hasAllSnippets(t, string(stdout), test.wantedStdoutSnippets); !ok {
+			continue
+		}
+		if ok := hasCFNStackWithClusterName(t, cfnClient, test.clusterName); !ok {
+			continue
+		}
+		if ok := hasClusterWithWantedConfig(t, ecsClient, &test); !ok {
+			continue
+		}
+
+		// Cleanup the created resources
+		after(cfnClient, ecsClient, test.clusterName)
+	}
+}
+
+// setup initializes all the clients needed by the upTest.
+func setup(t *testing.T) (cfnClient *cloudformation.CloudFormation, ecsClient *ecs.ECS) {
+	sess, err := session.NewSession()
+	if err != nil {
+		// Fail the upTest immediately if we won't be able to evaluate it
+		assert.FailNowf(t, "failed to create new session for upTest clients", "%v", err)
+	}
+
+	conf := aws.NewConfig()
+	cfnClient = cloudformation.New(sess, conf)
+	ecsClient = ecs.New(sess, conf)
+	return
+}
+
+// hasAllSnippets returns true if stdout contains each snippet in wantedSnippets, false otherwise.
+func hasAllSnippets(t *testing.T, stdout string, wantedSnippets []string) bool {
+	for _, snippet := range wantedSnippets {
+		if !assert.Contains(t, stdout, snippet) {
+			return false
+		}
+	}
+	return true
+}
+
+// hasCFNStackWithClusterName returns true if the CFN stack was created successfully, false otherwise.
+func hasCFNStackWithClusterName(t *testing.T, client *cloudformation.CloudFormation, clusterName string) bool {
+	resp, err := client.DescribeStacks(&cloudformation.DescribeStacksInput{
+		StackName: aws.String(stackName(clusterName)),
+	})
+	if err != nil {
+		return assert.Failf(t, "unexpected CloudFormation error during DescribeStacks", "wanted no errors, got %v", err)
+	}
+	if resp.Stacks == nil {
+		return assert.Fail(t, "stacks should not be nil")
+	}
+	if len(resp.Stacks) != 1 {
+		return assert.Failf(t, "did not receive only 1 stack", "wanted only one stack, got %d", len(resp.Stacks))
+	}
+	if *resp.Stacks[0].StackName != stackName(clusterName) {
+		return assert.Failf(t, "unexpected stack name", "wanted %s, got %s", stackName(clusterName), *resp.Stacks[0].StackName)
+	}
+	return true
+}
+
+// hasClusterWithWantedConfig returns true if the instances in the cluster are all active, have the required attributes
+// and resource, false otherwise.
+func hasClusterWithWantedConfig(t *testing.T, client *ecs.ECS, test *upTest) bool {
+	instances, err := containerInstances(t, client, test.clusterName, test.wantedClusterSize)
+	if err != nil {
+		assert.NoError(t, err)
+		return false
+	}
+	if !areInstancesActive(t, instances) {
+		return false
+	}
+	if !areInstancesWithAttributes(t, instances, test.wantedInstanceAttributes) {
+		return false
+	}
+	if !areInstancesWithResourceNames(t, instances, test.wantedInstanceResourceNames) {
+		return false
+	}
+	return true
+}
+
+// containerInstances returns the list of instances in the cluster.
+func containerInstances(t *testing.T, client *ecs.ECS, clusterName string, clusterSize int) ([]*ecs.ContainerInstance, error) {
```

Comment:
can we rename this "getContainerInstances" to reflect the action being performed?